apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/TreasuryStep/TreasuryStep.tsx

```diff
@@ -0,0 +1,204 @@
+import { useGetPersonalStakingStatistics } from '../../../../../StakePage/components/PersonalStakingStatistics/hooks/useGetPersonalStakingStatistics';
+import { useGetStakingStatistics } from '../../../../../StakePage/components/StakingStatistics/hooks/useGetStakingStatistics';
+import { useProposalContext } from '../../../../contexts/NewProposalContext';
+import {
+  ProposalCreationParameter,
+  ProposalCreationStep,
+} from '../../../../contexts/ProposalContext.types';
+import { Governor } from '../../NewProposalForm.types';
+import {
+  DEFAULT_PARAMETER,
+  GOVERNOR_VAULT_ADMIN_ADDRESS,
+  GOVERNOR_VAULT_OWNER_ADDRESS,
+  REQUIRED_VOTING_POWER,
+} from './TreasuryStep.constants';
+import { Parameter } from './components/Parameter/Parameter';
+
+type TreasuryStepProps = {
+  onPreview: () => void;
+  updateConfirmButtonState: (value: boolean) => void;
+};
+
+export const TreasuryStep: FC<TreasuryStepProps> = ({
+  onPreview,
+  updateConfirmButtonState,
+}) => {
+  const { setParameters, setStep, parameters, submit, governor, setGovernor } =
+    useProposalContext();
+  const [maxAmountError, setMaxAmountError] = useState(false);
+
+  const [governorOwner, setGovernorOwner] = useState('');
+  const [governorAdmin, setGovernorAdmin] = useState('');
+
+  const GOVERNOR_OPTIONS = useMemo(
+    () => [
+      {
+        value: governorAdmin,
+        label: t(translations.proposalPage.governorAdmin),
+      },
+      {
+        value: governorOwner,
+        label: t(translations.proposalPage.governorOwner),
+      },
+    ],
+    [governorAdmin, governorOwner],
+  );
+
+  const { votingPower } = useGetPersonalStakingStatistics();
+  const { totalVotingPower } = useGetStakingStatistics();
+
+  const isValidParameter = useCallback(
+    (parameter: ProposalCreationParameter) =>
+      isAddress(parameter?.treasuryStepExtraData?.recipientAddress || '') &&
+      Number(parameter?.treasuryStepExtraData?.amount) > 0 &&
+      !maxAmountError,
+    [maxAmountError],
+  );
+
+  const hasVotingPower = useMemo(() => {
+    if (votingPower && totalVotingPower) {
+      const requiredVotingPower = totalVotingPower * REQUIRED_VOTING_POWER;
+      return votingPower >= requiredVotingPower;
+    } else {
+      return false;
+    }
+  }, [votingPower, totalVotingPower]);
+
+  const isConfirmDisabled = useMemo(
+    () => !parameters.every(isValidParameter) || !hasVotingPower || !governor,
+    [parameters, isValidParameter, hasVotingPower, governor],
+  );
+
+  const handleAddClick = useCallback(() => {
+    const lastParameter = parameters[parameters.length - 1];
+    const nextIndex = (lastParameter?.treasuryStepExtraData?.index || 0) + 1;
+
+    const newParameter = {
+      ...DEFAULT_PARAMETER,
+      target:
+        lastParameter.target === GOVERNOR_VAULT_OWNER_ADDRESS ||
+        lastParameter.target === GOVERNOR_VAULT_ADMIN_ADDRESS
+          ? lastParameter.target
+          : GOVERNOR_VAULT_OWNER_ADDRESS,
```

Comment:
I believe checks here are not needed anymore and we can just have empty target or use lastParameter.target?